### ISSUE TITLE
Using fast_executemany = False on datetime2 column works, but fast_executemany = True fails

### ISSUE THREAD

**Issue by PrinsEdje80** [mkleehammer/pyodbc#540]
### Environment
- Python: Python 3.7.2 (64-bit)
- pyodbc: 4.0.26
- OS: Windows 10 Enterprise
- DB: SQL 2012 (11.0.7469.6)
- driver: ODBC Driver 17 for SQL Server (17.1.0.1)

### Issue
using executemany with `fast_executemany = True` gives an error `[22008] [Microsoft][ODBC Driver 17 for SQL Server]Datetime field overflow. Fractional second precision exceeds the scale specified in the parameter binding. (0) (SQLExecute)` whereas `fast_executemany = False` executes everything just fine.

### Expected result (from reproscript when working correctly)
Table with 5 values
| i | d |
| ---- | ---- |
| 1 | 2019-03-11 13:15:02.36 |
| 2 | 2019-03-11 13:15:02.36 |
| 3 | 2019-03-11 13:15:02.36 |
| 4 | 2019-03-11 13:15:02.36 |
| 5 | 2019-03-11 13:15:02.36 |

### Obtained result (how reproscript currently works)
Table with 3 values and an error `[22008] [Microsoft][ODBC Driver 17 for SQL Server]Datetime field overflow. Fractional second precision exceeds the scale specified in the parameter binding. (0) (SQLExecute)`
| i | d |
| ---- | ---- |
| 1 | 2019-03-11 13:15:02.36 |
| 2 | 2019-03-11 13:15:02.36 |
| 3 | 2019-03-11 13:15:02.36 |

### Reproscript
```python
server = 'myserver'
dbase = 'playground'
db_username = my_username
db_password = my_password

from os.path import basename

import datetime as dt
import pyodbc as po
connection_str = f"DRIVER={{ODBC Driver 17 for SQL Server}};SERVER={server};UID={db_username};PWD={db_password};MARS_Connection=No;APP={basename(__file__)};"

cnxn = po.connect(connection_str)
cnxn.autocommit = False

cr = cnxn.cursor()

table_sql = f"""IF OBJECT_ID('{dbase}.dbo.tst', 'U') IS NOT NULL DROP TABLE {dbase}.dbo.tst;
CREATE TABLE {dbase}.dbo.tst (
	i BIGINT NOT NULL IDENTITY(1,1)
	, d DATETIME2(2) NOT NULL
);"""

cr.execute(table_sql)
cr.commit()

data_sql = f"""INSERT INTO {dbase}.dbo.tst
	(d)
VALUES
	(?)
;"""

p_single = [dt.datetime.utcnow()]
p_multi  = [(dt.datetime.utcnow(), ), (dt.datetime.utcnow(), )]

cr.execute(data_sql, p_single)
cr.commit()

cr.fast_executemany = False
cr.executemany(data_sql, p_multi)
cr.commit()

cr.fast_executemany = True
cr.executemany(data_sql, p_multi)
cr.commit()
```

Apologies for not getting the tables right. Can't seem to get markdown to get it working...

**Comment by PrinsEdje80** [comment]
Same issue with ODBC Driver 17 for SQL Server (17.3.1.1)

**Comment by v-makouz** [comment]
This happens because pyODBC doesn't truncate the fraction part of datetime before sending it to the driver, so it correctly specifies precision as 2, but the fraction is still at full precision.

This is a pull-request that fixes the bug:
https://github.com/mkleehammer/pyodbc/pull/541

**Comment by DSLituiev** [comment]
which release is it supposed to be fixed?

I'm on '4.0.30', still getting `[ODBC Driver 17 for SQL Server]Datetime field overflow`

**Comment by matt-davis27** [comment]
Can confirm with @DSLituiev , I am using pyodbc 4.0.30 and still see the error when using the 17 Driver.

**Comment by gordthompson** [comment]
Fixed in #541
